src/timedata.cpp

```diff
@@ -40,16 +40,20 @@ static int64_t abs64(int64_t n)
     return (n >= 0 ? n : -n);
 }
 
+#define BITCOIN_TIMEDATA_MAX_SAMPLES 200
+
 void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
 {
     LOCK(cs_nTimeOffset);
     // Ignore duplicates
     static set<CNetAddr> setKnown;
+    if (setKnown.size() == BITCOIN_TIMEDATA_MAX_SAMPLES)
+        return;
```

Comment:
I didn't know the behavior in details: but does this mean if we have reached the `BITCOIN_TIMEDATA_MAX_SAMPLES ` of 200 we run into a risk of not detecting a significant time offset anymore?
Why not shift out the `CNetAddr` instance at index 0 if we have reached 200 (and still add the new one)?